pkg/agent/packetcapture/packetcapture_controller.go

```diff
@@ -464,10 +464,22 @@ func (c *Controller) performCapture(
 	}
 	defer pcapngWriter.Flush()
 	updateRateLimiter := rate.NewLimiter(rate.Every(captureStatusUpdatePeriod), 1)
-	packets, err := c.captureInterface.Capture(ctx, device, snapLen, srcIP, dstIP, pc.Spec.Packet)
-	if err != nil {
-		return false, err
+
+	var packets chan gopacket.Packet
+	if pc.Spec.Bidirection {
+		reqpackets, err1 := c.captureInterface.Capture(ctx, device, snapLen, srcIP, dstIP, pc.Spec.Packet)
+		respackets, err2 := c.captureInterface.Capture(ctx, device, snapLen, dstIP, srcIP, pc.Spec.Packet)
```

Comment:
@hangyan do we need 2 separate capture instances, or a single instance that can capture traffic in both directions?

Reply:
one is enough. The bpf code should be updated to support bidirection capture.

@AryanBakliwal You can use tcpdump to check the bpf coded it generated and update ours corresponding